presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/JoinMatcher.java

```diff
@@ -18,43 +18,61 @@
 import com.facebook.presto.metadata.Metadata;
 import com.facebook.presto.sql.planner.plan.JoinNode;
 import com.facebook.presto.sql.planner.plan.PlanNode;
+import com.google.common.collect.ImmutableSet;
 
 import java.util.List;
+import java.util.Set;
 
+import static com.facebook.presto.util.ImmutableCollectors.toImmutableSet;
 import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
 final class JoinMatcher
         implements Matcher
 {
     private final JoinNode.Type joinType;
-    private final List<AliasPair> equiCriteria;
+    private final List<ExpectedValueProvider<JoinNode.EquiJoinClause>> equiCriteria;
 
-    JoinMatcher(JoinNode.Type joinType, List<AliasPair> equiCriteria)
+    JoinMatcher(JoinNode.Type joinType, List<ExpectedValueProvider<JoinNode.EquiJoinClause>> equiCriteria)
```

Comment:
That's all anybody's using it for right now. SemiJoins are handled with SemiJoinMatcher